src/lib/uri/uri.h

```diff
@@ -154,6 +154,14 @@ uri_escape(const char *src, size_t src_size, char *dst,
 size_t
 uri_unescape(const char *src, size_t src_size, char *dst, bool decode_plus);
 
+/** Determine if uris refer to the same host:service or unix socket path. */
+bool
+uri_equals(const struct uri *a, const struct uri *b);
```

Comment:
@locker, what is the most canonic rule? `_equals()` or `_is_equal()`? I see both versions.

@sergepetrenko - the commit message mentions `uris_are_equal()` which is probably outdated. In the current version I see `uri_equals()`. Besides, it doesn't look like uris are equal. You compare only the host and port. So probably it should be called `_addr_is_equal()` or alike.

Reply:
> @locker, what is the most canonic rule? `_equals()` or `_is_equal()`? I see both versions.

`_is_equal()`. I see the only place where we use `_equals()` - [`tt_bitset_index_expr_equals()`](https://github.com/tarantool/tarantool/blob/9fcafbd14b962d9f844a9d114518fa690753f276/src/lib/bitset/index.h#L211), but it's ancient code and it doesn't actually compare two objects of the same type.

Reply:
Thanks!

Reply:
Changed `uri_equals()` to `uri_addr_is_equal()`:
<details><summary>diff</summary>

```diff
diff --git a/src/lib/uri/uri.c b/src/lib/uri/uri.c
index 07551cc53..7e9ac10f5 100644
--- a/src/lib/uri/uri.c
+++ b/src/lib/uri/uri.c
@@ -469,7 +469,7 @@ fields_are_equal(const char *a, const char *b)
 }
 
 bool
-uri_equals(const struct uri *a, const struct uri *b)
+uri_addr_is_equal(const struct uri *a, const struct uri *b)
 {
 	/*
 	 * Either service or path will be NULL depending on whether this is a
diff --git a/src/lib/uri/uri.h b/src/lib/uri/uri.h
index b068592b6..78e453fed 100644
--- a/src/lib/uri/uri.h
+++ b/src/lib/uri/uri.h
@@ -156,7 +156,7 @@ uri_unescape(const char *src, size_t src_size, char *dst, bool decode_plus);
 
 /** Determine if uris refer to the same host:service or unix socket path. */
 bool
-uri_equals(const struct uri *a, const struct uri *b);
+uri_addr_is_equal(const struct uri *a, const struct uri *b);
 
 /** Check if a uri is empty. */
 bool
diff --git a/test/unit/uri.c b/test/unit/uri.c
index 97386d85f..e97166408 100644
--- a/test/unit/uri.c
+++ b/test/unit/uri.c
@@ -148,7 +148,7 @@ struct uri_equal_expected {
 };
 
 static void
-test_equals(void)
+test_addr_is_equal(void)
 {
 	struct uri_equal_expected test_pairs[] = {
 		{NULL, NULL, true},
@@ -177,9 +177,9 @@ test_equals(void)
 		   src_a ? src_a : "NULL");
 		ok(uri_create(&uri_b, src_b) == 0, "uri_create(%s)",
 		   src_b ? src_b : "NULL");
-		is(uri_equals(&uri_a, &uri_b), is_equal, "%s %s equal to %s",
-		   src_a ? src_a : "NULL", is_equal ? "is" : "isn't",
-		   src_b ? src_b : "NULL");
+		is(uri_addr_is_equal(&uri_a, &uri_b), is_equal,
+		   "%s %s equal to %s", src_a ? src_a : "NULL",
+		   is_equal ? "is" : "isn't", src_b ? src_b : "NULL");
 	}
 	check_plan();
 	footer();
@@ -833,7 +833,7 @@ main(void)
 	test_copy_empty();
 	test_move_sample();
 	test_move_empty();
-	test_equals();
+	test_addr_is_equal();
 	test_string_uri_with_query_params_parse();
 	test_string_uri_set_with_query_params_parse();
 	test_invalid_string_uri_set();
```
</details>